Core/NoOpRunner.Core/Shapes/Player.cs

```diff
@@ -128,6 +132,27 @@ public void HandleKeyPress(KeyPress key, WindowPixel[,] gameScreen)
             }
         }
 
+        public void DoDamage(int damage)
+        {
+            CurrentHealth -= damage;
+
+            if(CurrentHealth < 1)
+            {
+                CurrentHealth = 0;
+                // Stop the game
+            }
+        }
+
+        public void DoHeal(int heal)
+        {
+            CurrentHealth += heal;
+
+            if (CurrentHealth > MaxHealth)
+            {
+                CurrentHealth = MaxHealth;
+            }
+        }
+
```

Comment:
As per other comments, leave the functions only on Player (so here) and merge them into one.

Reply:
Fixed